lib/FirmwareModule.js

```diff
@@ -106,7 +111,11 @@ FirmwareModule.prototype = extend(FirmwareModule.prototype, {
 
 		var HalDependencyResolver = require('./HalDependencyResolver');
 		var resolver = new HalDependencyResolver();
-		var invalidModules = resolver.solveFirmwareModule(deviceModules, this.dependencies[0]);
+
+		for (let dep of this.dependencies) {
+			const invalidDependencyModules = resolver.solveFirmwareModule(deviceModules, dep);
+			invalidModules.push.apply(invalidModules, invalidDependencyModules.map(m => new FirmwareModule(m)));
```

Comment:
`invalidModules = invalidModules.concat(invalidDependencyModules)`?

Reply:
We can't use `concat` since it doesn't modify in place and returns a new array.

Reply:
ah, indeed, the array object is provided by the calling code